nixos/modules/installer/sd-card/sd-image-mips64el.nix

```diff
@@ -0,0 +1,35 @@
+# To build, use:
+# nix-build nixos -I nixos-config=nixos/modules/installer/sd-card/sd-image-mips64el.nix -A config.system.build.sdImage
+{ config, lib, pkgs, ... }:
+
+{
+  imports = [
+    ../../profiles/base.nix
+    ./sd-image.nix
+  ];
+
+  boot.loader = {
+    grub.enable = false;
+    generic-extlinux-compatible = {
+      enable = true;
+    };
+  };
+
+  services.xserver.enable = lib.mkForce false;
+  services.xserver.libinput.enable = lib.mkForce false;
+
+  boot.consoleLogLevel = lib.mkDefault 7;
+  boot.kernelParams = [ "console=ttyS0,115200" ];
+  boot.initrd.includeDefaultModules = false;
+
+  nixpkgs.buildPlatform = builtins.currentSystem;
+  nixpkgs.hostPlatform = lib.systems.examples.octeon;
```

Comment:
I'm not sure I can review anything else, but there's at least this.

This would need to use `mips64el` (since it's the name of the file) or rename the file for `octeon`. I'd prefer targeting the generic `mips64el` and users needing octeon can provide a host platform.

But actually, with a bit more thought, SD/ISO builds should be made assuming native builds. Cross-compiling usage is done externally through configuring platforms. Look at all the other existing sd image configs. So actually removing the lines is the correct resolution here.

```suggestion
```

Reply:
@roberth [is insisting](https://github.com/NixOS/nixpkgs/pull/246690#pullrequestreview-1560011489) that it be done this way -- through module parameters.

Perhaps one of you can convince the other.

Reply:
(I left a comment on the other PR)

There is no convincing to do, and we agree, and this comment here agrees with the idea that cross-compilation is configured through the NixOS configuration.

Reply:
This module is a "profile" module that's specifically about mips64el, so it does make sense to set a default `hostPlatform`, but `buildPlatform` is not specific to mips64el hosts and therefore shouldn't be set here.